hedera-node/src/main/java/com/hedera/services/state/migration/MigrationRecordsManager.java

```diff
@@ -224,13 +270,91 @@ private void publishContractFreeAutoRenewalRecords() {
                         });
     }
 
-    @VisibleForTesting
-    void setSideEffectsFactory(Supplier<SideEffectsTracker> sideEffectsFactory) {
-        this.sideEffectsFactory = sideEffectsFactory;
+    private void publishTraceabilityMigrationRecords() {
+        final var contractStorageMap = contractStorage.get();
+        final var allContractsStateChangesBuilder = ContractStateChanges.newBuilder();
+        accounts.get()
+                .forEach(
+                        (id, account) -> {
+                            if (account.isSmartContract() && !account.isDeleted()) {
```

Comment:
This method is hard to read due to the unnecessary indentation and nested loop/conditionals. It would benefit from being further split into other functions. At minimum, simplify the forEach statement and use fail fast logic to reduce nesting.

```suggestion
        accounts.get().forEach((id, account) -> {
            if (!account.isSmartContract()) {
                return;
            }
```

Reply:
Refactored and extracted methods for bytecode and state changes generation. 
I'm afraid the `AOSP` style being applied by `spotless` inserts new lines on questionable places at times 😄